modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/PlannerHelper.java

```diff
@@ -180,7 +180,7 @@ public static IgniteRel optimize(SqlNode sqlNode, IgnitePlanner planner) {
                 LOG.debug(planner.dump());
             }
 
-            if (ex instanceof CannotPlanException) {
+            if (ex instanceof CannotPlanException || ex instanceof SqlException) {
```

Comment:
why do u need such a change ? it work properly without it, also as for me you need to append additional test with : @WithSystemProperty(key = "FAST_QUERY_OPTIMIZATION_ENABLED", value = "false")
and also rename initial issue

Reply:
Because this change fixes described problem.
Please, try to run `ItCastToIntTest` WITHOUT this change and you'll see that the `explicitCastOfLiteralsOnSelectWithOverflow` test will fail.

Reply:
>  you need to append additional test with : @WithSystemProperty(key = "FAST_QUERY_OPTIMIZATION_ENABLED", value = "false")

Done

> rename initial issue

I don't get why and how to rename, this issue is reproduced even fast query optimization is turned off